### PR TITLE
Upgrade to eslint 8

### PR DESCRIPTION
**Pull request by dennis-chen-vanta**
Do a major version bump of eslint, bringing us to 8. Update corresponding dependencies and dependees as necessary.
[sc-40936]

### CODE REVIEW

**Review by ostrowr — APPROVED** [review]
lgtm!

File: lib/rules/common-absolute-import.ts [hunk 1]
```diff
@@ -12,7 +12,6 @@ const rule = ESLintUtils.RuleCreator(
   meta: {
     fixable: "code",
     docs: {
-      category: "Possible Errors",
```
Comment (dennis-chen-vanta) [inline lib/rules/common-absolute-import.ts:15]: This attribute was removed in https://github.com/typescript-eslint/typescript-eslint/pull/3800 , so needed to mirror that change when upgrading experimental eslint utils.

File: package-lock.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-plugin-vanta",
-  "version": "1.0.0",
+  "version": "0.0.1",
```
Comment (dennis-chen-vanta) [inline package-lock.json:3]: I couldn't figure out if I was doing something wrong here or have something misconfigured locally that caused the version number to get bumped down like this. Any ideas on this?

Comment (dennis-chen-vanta) [reply package-lock.json:3]: Nope, this was a result of the version in package.json being updated in the past, without `npm install` being rerun at that point. This is safe and brings package.json and package-lock.json back into sync.

**Pull request merged**